operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/GroupVersionKind.java

```diff
@@ -2,12 +2,15 @@
 
 import java.util.Objects;
 
+import io.fabric8.kubernetes.api.Pluralize;
 import io.fabric8.kubernetes.api.model.HasMetadata;
 
 public class GroupVersionKind {
   private final String group;
   private final String version;
   private final String kind;
+  private final String plural;
+  private final String apiVersion;
```

Comment:
We don't need explicit API Version, it is "group/version".

Reply:
This is to avoid re-computing it all the time.

Reply:
I don't think we should do that. It you check this is only used when creating the informer, we should not do such optimizations. It would, worth to check but from java 9 there are quite clever optimizations on String concatenation, probably making such optimizations unnecessary.

Reply:
Let's take a look on this on opposite direction, this could also be seen also as premature optimization. It brings complexity to the code. (Althoug not significant for sure). If you can show this significantly increases any performance, it's fine. But I really don't think we should do such micro optimizations, especially in case of string concatenations, that are not repeadetly used. 

I mean we could argue that this also has negative side on memory usage. 

At the end it is more about complexity of the code in this case.

Reply:
But I might be wrong :)  What do you think?  maybe @xstefank @andreaTP @manusa

Reply:
> Let's take a look on this on opposite direction, this could also be seen also as premature optimization. It brings complexity to the code. (Althoug not significant for sure). If you can show this significantly increases any performance, it's fine. But I really don't think we should do such micro optimizations, especially in case of string concatenations (that is optimized out on JVM nicely) , that are not repeadetly used.

It doesn't bring any complexity: the code that used to be in the getter is used in the constructor instead and, to the contrary, it simplifies both equals and hashCode (because you can just check apiVersion instead of needing to check both group and version), also "optimizing" their execution. So the complexity argument is just wrong, imo.

> We could argue that this also has negative side on memory usage.

That, I agree with but, in this case, I'm sure there are many other spots where we could save more memory than just saving an extra string here.

Reply:
> That, I agree with but, in this case, I'm sure there are many other spots where we could save more memory than just saving an extra string here. 

We could do performance optimizations on other spots too, this is not a valid argument.

Reply:
IMHO, it could be in the field; this won't take that much space. While I understand Attila's arguments, I more often see this kind of computation cache like Chris is proposing.

Reply:
oki, I'm fine with that. Might be good to comment on it, that is for performance reasons.